### PR TITLE
[fix #2731] Allow feature flags to be turned on/off

### LINKED ISSUE #2731: Allow feature flags to be turned on/off
### User Story

As a developer, I want to turn on/off feature flags from a simple developer screen

### Description

Some features part of a developer release are hidden behind a statically set flag. As a developer I want to be able to change the value of those flags from a simple screen. Restarting would then allow me to see relevant changes.

See [config.cljs](https://github.com/status-im/status-react/blob/develop/src/status_im/utils/config.cljs) for more details about how those flags are currently used.

### PR DESCRIPTION
**Pull request by t4sk**
Allow feature flags to be turned on/off from a developer screen

Fixes #2731 

### Summary:
TODO:
- [x] Create developer settings screen
- [x] On app init, load `.env` config to `db`
- [ ] Also load config from local storage and override config in `db`
- [ ] Upon updating a setting, update `db` and store it to local storage
- [ ] Wherever [config functions](https://github.com/status-im/status-react/blob/develop/src/status_im/utils/config.cljs) are called, update logic to get value from `db`
- [ ]  "reinject" flags into native code or remove it from the list
- [ ] Include developer settings screen only in dev environment
- [ ] In development, add a link to navigate to developer screen

Sample Screen:
screenshot_15154201001

### Steps to test:

status: wip

### CODE REVIEW

**Comment by oskarth** [comment]
Great start @t4sk! Will review more thoroughly later.

What do you think the best way to deal with flags that require app to be 'restarted' (init-whisper, say)? Can these be set and then prompt the user/auto restart or something like this?

**Comment by rasom** [comment]
> Can these be set and then prompt the user/auto restart or something like this?

in this case we also would need to store them in db and it's not clear at which point we should fetch them and apply. 

Also i have doubts that we can "reinject" flags which are used in native code, and this means if the same flag is used in both cljs and native code we can end up in some inconsistent state (like we set JSC_ENABLED=1, this changes some behavior in cljs but it's still JSC_ENABLED=0 in Java). So we need to make sure that we can pass and apply these flags in native code, otherwise remove them from this list.

**Comment by t4sk** [comment]
### Problem
React Native only supports an async local storage. All the functions which depend on feature flags will need to be called AFTER the configurations has been loaded. It seems like a lot of code changes.

### Solution
Save the configurations in a json file, `require` it and merge it with `config`. 

### Pros
- `status-im.utils.config` will have the updated values available during boot. 
- less code change

### Example code:

`require` `.env.tmp.json` in [status-im.utils.config](https://github.com/status-im/status-react/blob/develop/src/status_im/utils/config.cljs). This file will store the updated settings.  Update the file when the developer changes the setting.
```Clojure
(def config (js->clj (.-default rn-dependencies/config) :keywordize-keys true))

(def tmp-config (json->clj (slurp ".env.tmp.json")))

;; merge config and tmp-config to get the updated values
```

Any feedback? Thanks in advance

**Comment by jeluard** [comment]
@t4sk Would this address @rasom comment about native code?

**Comment by t4sk** [comment]
@jeluard Probably not. I would have to do some googling to see if it is possible to reinject the flags into native code.

**Comment by jeluard** [comment]
@t4sk So how about we explicitly remove those from the list? There is at least `TESTFAIRY_ENABLED`, `ETHEREUM_DEV_CLUSTER` and `JSC_ENABLED`.

**Comment by t4sk** [comment]
@jeluard Definitely, I will do a full text search and check which flags are used in the native code.

**Comment by oskarth** [comment]
Still WIP, moving back to CONTRIBUTOR.

**Comment by oskarth** [comment]
@t4sk Any luck with this?

File: src/status_im/ui/screens/dev/db.cljs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+(ns status-im.ui.screens.dev.db
+  (:require-macros [status-im.utils.db :refer [allowed-keys]])
+  (:require [cljs.spec.alpha :as spec]))
+
+(spec/def :dev/testfairy-enabled? boolean?)
```
Comment (oskarth) [inline src/status_im/ui/screens/dev/db.cljs:5]: Thanks for specs btw!

**Comment by t4sk** [comment]
I am not sure how to save the config (write to file). 

Things I tried that didn't work:
- use `react-native-fs`, but it's not helpful here since it writes to a file  inside the mobile device.
- using `clojure/spit`, similar to how `status-im.utils.slurp` is being used, crashes the app.

Should I launch a node server and send a http request to write to a file? It feels hacky.

**Comment by oskarth** [comment]
^ @rasom Do you have any thoughts on this?

**Comment by oskarth** [comment]
@t4sk Can we break this PR into smaller changes perhaps? So we can merge it quicker. As long as it brings some utility for some flags it is good enough IMO. Closing this PR due to inactivity / being too big.

**Pull request closed without merging**